docs/root/configuration/overview/examples.rst

```diff
@@ -302,3 +302,20 @@ The management server could respond to EDS requests with:
             socket_address:
               address: 127.0.0.2
               port_value: 1234
+
+Special YAML usage
+~~~~~~~~~~~~~~~~~~
+
+When loading YAML configuration, the Envoy loader will interpret map keys tagged with !ignore
+specially, and omit them entirely from the native configuration tree. Ordinarily, the YAML stream
+must adhere strictly to the proto schemas defined for Envoy configuration. This allows content to
+be declared that is explicitly handled as a non-represented type.
+
+One potential use for this mechanism is to avoid extra interpolation or pre-processing overhead when
+dynamically generating YAML configuration for Envoy. Dynamic YAML anchors may be provided at the head
+of the stream, followed by a "template" that aliases those values.
```

Comment:
This is a bit confusing, can you give an example of how it is supposed to work? It seems you want some kind of templating, but unclear why parameters need to be ignored to do this.

Reply:
@htuch There's an example in the attached yaml file. I hoped it would be illustrative of the intent here.

The reason parameters need to be ignored is that otherwise YAML anchors must be declared in-line to the structure of the document. (For instance, if you wanted to template some TLS parameters, your first usage would have to be literally declared in the first cluster, and subsequent clusters could reference it.) By allowing nodes to be defined that will be excluded from the deserialized config structure, we can separate a "pure" template containing only aliases, from declarations of dynamic content, containing only anchors. In both the attached example, and in Envoy Mobile, we intend to leverage this by passing yaml effectively in two chunks - one that's ignored by deserialization and contains dynamic values, and one that contains a fixed template. By separating them, we can avoid an extra interpolation or preprocessing step.

By way of another example, I've seen deployments rely on templating tools like jinja2 to template static/bootstrap config in the past. With this setup, they could avoid the extra tool and preprocessing step, and simply output appropriately serialized dynamic content followed by their template -- the loader will handle the rest.

Please let me know if that's still unclear, or if you have thoughts on how I can improve upon the description or example.

Reply:
Keeping things simple, which is always good for ESL etc, I would just say "This lets you split your file into two parts: one in which we have YAML content not subject to parsing according to the schema and another part that is parsed. YAML anchors in the first part may be referenced by the second part."

Or something very simple like that. It honestly took me some time to parse "
 extra interpolation or pre-processing overhead whendynamically generating YAML configuration for Envoy". Like, "extra" compared to what? Who is interpolating where? What is the preprocessor and how?

Sorry to be picky, I'm just trying to feedback to better docs here.

Reply:
No worries, I never feel very certain that I'm explaining things clearly, so this is helpful feedback. I'll take another crack at the wording with your suggestion. Thank you!

Reply:
@htuch I incorporated your suggestion pretty directly. Please let me know if that seems clearer.

Reply:
LGTM